src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,116 @@
+import { html, LitElement, PropertyValues, nothing } from "lit";
+import { property } from "lit/decorators.js"
+import { computePosition, arrow, offset, MiddlewareData } from "@floating-ui/dom";
+import "../icons/index.js";
+import "../icon-button";
+
+export class Tooltip extends LitElement {
+  @property({ type: String, reflect: true }) placement:
+  | 'top'
+  | 'top-start'
+  | 'top-end'
+  | 'right'
+  | 'right-start'
+  | 'right-end'
+  | 'bottom'
+  | 'bottom-start'
+  | 'bottom-end'
+  | 'left'
+  | 'left-start'
+  | 'left-end' = 'top';
+  @property({ type: Boolean, reflect: true }) dismissible: boolean = true;
+  @property({ type: String, reflect: true }) width: string = "0";
+  @property({ type: String, reflect: true }) arrowOffset: string = "";
```

Comment:
Since we are defining a HTML tag, it's better to define our attributes in kebab case:

```suggestion
  @property({ type: String, reflect: true, attribute: 'arrow-offset' }) arrowOffset: string = "";
```

also please update story